xarray/backends/api.py

```diff
@@ -743,19 +763,26 @@ def save_mfdataset(datasets, paths, mode='w', format=None, groups=None,
                          'save_mfdataset')
 
     writer = ArrayWriter()
-    stores = [to_netcdf(ds, path, mode, format, group, engine, writer)
+    stores = [to_netcdf(ds, path, mode, format, group, engine, writer,
+                        compute=compute)
               for ds, path, group in zip(datasets, paths, groups)]
+
     try:
-        writer.sync()
+        delayed = writer.sync(compute=compute)
+        if not compute:
+            import dask
+            # or should this return an iterable of delayed objects
+            # (one for each store)
+            return dask.delayed(_finalize_stores)(stores, delayed)
```

Comment:
I'm open to ideas from folks on this one. Do we want to return a list of delayed objects or a single delayed object when using this option with `save_mfdataset`?